### PR TITLE
export codegen tests for internal use

### PR DESCRIPTION
**Pull request by iwahbe**
# Description

Fixes # (issue)

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

**Review by mikhailshilkov — APPROVED** [review]

File: pkg/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# pulumi/pkg
+
+The files in `pulumi/pkg` make up the internals of the Pulumi CLI. Unlike `pulumi/sdk`, all files are considered "internal", and
+we maintain no backwards compatibility guaranty on this module.
```
Comment (t0yv0) [inline pkg/README.md:4]: We could frame this in a positive way.

> while pulumi/sdk maintains strict backwards compatibility guarantees, code under pkg/ is handled more informally, while we avoid making breaking changes unnecessarily they might happen when they make sense.

**Review by t0yv0 — APPROVED** [review]

**Comment by t0yv0** [comment]
Before merging give me an hour or so to try it out in my repo, I will have feedback or confirmation that it works.

**Comment by t0yv0** [comment]
So the first problem is that although funcs and structs are Public, they are in an `internal` package which Go toolchain does not let us cross-reference from other packages, as in:

```
	gen_test.go:10:2: use of internal package github.com/pulumi/pulumi/pkg/codegen/internal/test not allowed
FAIL	github.com/pulumi/pulumi-java/pkg/codegen/jvm [setup failed]
```

**Comment by iwahbe** [comment]
> So the first problem is that although funcs and structs are Public, they are in an `internal` package which Go toolchain does not let us cross-reference from other packages, as in:
> 
> ```
> 	gen_test.go:10:2: use of internal package github.com/pulumi/pulumi/pkg/codegen/internal/test not allowed
> FAIL	github.com/pulumi/pulumi-java/pkg/codegen/jvm [setup failed]
> ```

I have renamed `codegen/internal` to `codegen/testing`.

**Pull request merged**